packages/core/src/App/Containers/AccountSwitcherWallet/account-switcher-wallet-item.tsx

```diff
@@ -39,6 +41,8 @@ export const AccountSwitcherWalletItem = observer(
         const icon_type = is_virtual ? 'demo' : currency_config?.type;
         const is_selected = is_active || linked_to?.some(account => account.loginid === active_account?.loginid);
 
+        const show_badge = is_malta_wallet || is_virtual;
```

Comment:
same here

Reply:
@mahdiyeh-deriv I'm not sure it's a good idea for this component (`AccountSwitcherWalletItem`), because this component needs to show every item in dropdown and we call it in `AccountSwitcherWallet` like this:

```
{dtrade_account_wallets?.map(account => (
                    <AccountSwitcherWalletItem
                        key={account.dtrade_loginid}
                        account={account}
                        closeAccountsDialog={closeAccountsDialog}
                    />
                ))}
```
So we just pass account and define to show or not the badge inside of the component. In your suggestion we need to add the same logic in AccountSwitcherWallet inside of the map method. Do you think is it better?
I understand what you mean, you want a dummy component as possible, but in my opinion this logic related to only this component. Also, we already have a lot of logic to define app_icon, icon_type, selected item or not, so I don't see any problem to add the logic to define to show the badge or not, because it depends on account which we already pass to the component.

Reply:
It's fine, keep it here and remove form the children component